DatadogCore/Sources/Core/Storage/FilesOrchestrator.swift

```diff
@@ -61,6 +73,16 @@ internal class FilesOrchestrator: FilesOrchestratorType {
         self.dateProvider = dateProvider
         self.telemetry = telemetry
         self.metricsData = metricsData
+
+        #if canImport(UIKit)
+        self.subscription = applicationStatePublisher.subscribe { [weak self] appStateHistory in
+            self?.inBackground = appStateHistory.currentSnapshot.state == .background
+        }
+        #endif
```

Comment:
The [`ApplicationStatePublisher`](https://github.com/DataDog/dd-sdk-ios/blob/c645cce1fd43256c4acb67e4e7462e27e815eca5/DatadogCore/Sources/Core/Context/ApplicationStatePublisher.swift#L19-L22) comes with a dispatch queue. If used within `FilesOrchestrator`, it will be (pessimistically) adding 1 thread per feature to manage this subscription. This won't be scalable.

In fact, this is V1-like approach, which was modernised in V2. In V2 `AppStateHistory` is already available in [`DatadogContext.applicationStateHistory`](https://github.com/DataDog/dd-sdk-ios/blob/c645cce1fd43256c4acb67e4e7462e27e815eca5/DatadogInternal/Sources/Context/DatadogContext.swift#L73-L74), so there is no need to subscribe again and consume more resources.

We need `currentSnapshot.state` in batch removal flow, hence we can capture it from here and pass downstream to `sendBatchDeletedMetric(batchFile:deletionReason:)`:
https://github.com/DataDog/dd-sdk-ios/blob/c645cce1fd43256c4acb67e4e7462e27e815eca5/DatadogCore/Sources/Core/Upload/DataUploadWorker.swift#L60

Reply:
Great input! I wasn't really sure about this one.

Thanks for the additional context - I'll make use of it 🙌